tango/integrations/beaker/workspace.py

```diff
@@ -331,25 +335,70 @@ def register_run(self, targets: Iterable[Step], name: Optional[str] = None) -> R
 
         return Run(name=cast(str, name), steps=steps, start_date=run_dataset.created)
 
-    def registered_runs(self) -> Dict[str, Run]:
-        import concurrent.futures
-
-        runs: Dict[str, Run] = {}
+    def search_registered_runs(
+        self,
+        *,
+        sort_by: WorkspaceSort = WorkspaceSort.START_DATE,
+        sort_descending: bool = True,
+        match: Optional[str] = None,
+        limit: Optional[int] = None,
```

Comment:
Doesn't limit have to be a slice object, so you can request a range? Or is that how you use `cursor` below? In that case we should change the names. Cursors are something else.

Reply:
`cursor` and `limit` can be used together to get a slice. I named them this way to mirror Beaker's API, but we can call them something else.

Reply:
I think those names are wrong in Beaker then too. That's not what a cursor is.

It's most Pythonic to do what the `slice` object does. `start`, `end`, `step`. And we don't really need `step` here.

Reply:
Okay, I'll make that change

Reply:
Well, if we're not using `step`, why not just rename `cursor` to `start` and `limit` to `stop`. Like, what's the benefit of taking an actually `slice` object? I don't think it's Pythonic to have a method that takes a slice object, other than the magic method `__getitem__`.

Reply:
4ee9ba1